### PR TITLE
Java package support

### PR DESCRIPTION
**Pull request by sundbry**
Reproduces and fixes issue #58 

```
Testing protojure.plugin.core-test

FAIL in (java-package-test) (core_test.clj:143)
Check to ensure that code generated with java_package uses the correct service name
expected: (= "kitchen.toaster.Toaster" toaster-client/Toaster-service-name)
  actual: (not (= "kitchen.toaster.Toaster" "com.example.kitchen.toaster.Toaster"))

FAIL in (java-package-test) (core_test.clj:144)
Check to ensure that code generated with java_package uses the correct service name
expected: (= "kitchen.toaster.Toaster" toaster-server/Toaster-service-name)
  actual: (not (= "kitchen.toaster.Toaster" "com.example.kitchen.toaster.Toaster"))
  ```

### CODE REVIEW

**Comment by sundbry** [comment]
From toaster.proto
```proto
syntax = "proto3";
package kitchen.toaster;
option java_package = "com.example.kitchen.toaster";
...
```

Client before:
```clj
;;;----------------------------------------------------------------------------------
;;; Generated by protoc-gen-clojure.  DO NOT EDIT
;;;
;;; GRPC com.example.kitchen.toaster.Toaster Client Implementation
;;;----------------------------------------------------------------------------------
(ns com.example.kitchen.toaster.Toaster.client
  (:require [com.example.kitchen.toaster :refer :all]
            [clojure.core.async :as async]
            [protojure.grpc.client.utils :refer [send-unary-params invoke-unary]]
            [promesa.core :as p]
            [protojure.grpc.client.api :as grpc]))

;-----------------------------------------------------------------------------
; GRPC Client Implementation
;-----------------------------------------------------------------------------

(def Toaster-service-name "com.example.kitchen.toaster.Toaster")

(defn Toast
  ([client params] (Toast client {} params))
  ([client metadata params]
  (let [input (async/chan 1)
        output (async/chan 1)
        desc {:service "com.example.kitchen.toaster.Toaster"
              :method  "Toast"
              :input   {:f com.example.kitchen.toaster/new-ToastRequest :ch input}
              :output  {:f com.example.kitchen.toaster/pb->ToastResponse :ch output}
              :metadata metadata}]
    (-> (send-unary-params input params)
        (p/then (fn [_] (invoke-unary client desc output)))))))
```

Client after:

```clj
;;;----------------------------------------------------------------------------------
;;; Generated by protoc-gen-clojure.  DO NOT EDIT
;;;
;;; GRPC com.example.kitchen.toaster.Toaster Client Implementation
;;;----------------------------------------------------------------------------------
(ns com.example.kitchen.toaster.Toaster.client
  (:require [com.example.kitchen.toaster :refer :all]
            [clojure.core.async :as async]
            [protojure.grpc.client.utils :refer [send-unary-params invoke-unary]]
            [promesa.core :as p]
            [protojure.grpc.client.api :as grpc]))

;-----------------------------------------------------------------------------
; GRPC Client Implementation
;-----------------------------------------------------------------------------

(def Toaster-service-name "kitchen.toaster.Toaster")

(defn Toast
  ([client params] (Toast client {} params))
  ([client metadata params]
  (let [input (async/chan 1)
        output (async/chan 1)
        desc {:service "kitchen.toaster.Toaster"
              :method  "Toast"
              :input   {:f com.example.kitchen.toaster/new-ToastRequest :ch input}
              :output  {:f com.example.kitchen.toaster/pb->ToastResponse :ch output}
              :metadata metadata}]
    (-> (send-unary-params input params)
        (p/then (fn [_] (invoke-unary client desc output)))))))
```

Server before:
```
;;;----------------------------------------------------------------------------------
;;; Generated by protoc-gen-clojure.  DO NOT EDIT
;;;
;;; GRPC com.example.kitchen.toaster.Toaster Service Implementation
;;;----------------------------------------------------------------------------------
(ns com.example.kitchen.toaster.Toaster.server
  (:require [com.example.kitchen.toaster :refer :all]
))

;-----------------------------------------------------------------------------
; GRPC Toaster
;-----------------------------------------------------------------------------
(defprotocol Service
  (Toast [this param]))

(def Toaster-service-name "com.example.kitchen.toaster.Toaster")

(defn- Toast-dispatch
  [ctx request]
  (Toast ctx request))

(def ^:const rpc-metadata
  [{:pkg "com.example.kitchen.toaster" :service "Toaster" :method "Toast" :method-fn Toast-dispatch :server-streaming false :client-streaming false :input pb->ToastRequest :output new-ToastResponse}])
```

Server after:
```clj
;;;----------------------------------------------------------------------------------
;;; Generated by protoc-gen-clojure.  DO NOT EDIT
;;;
;;; GRPC com.example.kitchen.toaster.Toaster Service Implementation
;;;----------------------------------------------------------------------------------
(ns com.example.kitchen.toaster.Toaster.server
  (:require [com.example.kitchen.toaster :refer :all]
))

;-----------------------------------------------------------------------------
; GRPC Toaster
;-----------------------------------------------------------------------------
(defprotocol Service
  (Toast [this param]))

(def Toaster-service-name "kitchen.toaster.Toaster")

(defn- Toast-dispatch
  [ctx request]
  (Toast ctx request))

(def ^:const rpc-metadata
  [{:pkg "kitchen.toaster" :service "Toaster" :method "Toast" :method-fn Toast-dispatch :server-streaming false :client-streaming false :input pb->ToastRequest :output new-ToastResponse}])
```

**Comment by sundbry** [comment]
@ghaskins I think we should bump the minor version number here for this and make a note in case any users are relying on the buggy behavior in production.

File: project.clj [hunk 1]
```diff
@@ -13,6 +13,7 @@
             [lein-cljfmt "0.8.0" :exclusions [org.clojure/clojure]]]
   :dependencies [[org.clojure/clojure "1.10.3"]
                  [io.github.protojure/core "2.0.1"]
+                 [io.github.protojure/grpc-client "2.0.1"]
```
Comment (ghaskins) [inline project.clj:16]: I think the latest is v2.0.9 so might as well bump these up, while you are adding a new one.  I know you were just matching the core 2.0.1

**Review by ghaskins — CHANGES_REQUESTED** [review]
Code looks good.  One minor nit re: versions commented inline.  Also, please add yourself to the CONTRIBUTORs.  Ty for the patch!

File: project.clj (hunk 1 above)
Comment (sundbry) [reply project.clj]: Done & Done!

**Review by ghaskins — APPROVED** [review]

**Comment by ghaskins** [comment]
Thanks @sundbry.  Ill bump the version and release with a note, as you suggested.

**Comment by ghaskins** [comment]
@sundbry https://github.com/protojure/protoc-plugin/releases/tag/v2.1.0

thanks again

**Comment by sundbry** [comment]
Thank you too for releasing @ghaskins!

**Pull request merged**